web-components/src/components/form/BasketTakeForm.tsx

```diff
@@ -0,0 +1,228 @@
+    let errors: FormikErrors<CreditTakeFormValues> = {};
+
+    if (!values.amount) {
+      errors.amount = requiredMessage;
+    } else if (Math.sign(values.amount) !== 1) {
+      errors.amount = invalidAmount;
+    } else if (values.amount > availableTradableAmount) {
+      errors.amount = insufficientCredits;
+    }
+
+    // Retire form validation (optional subform)
+    if (values.retireOnTake) {
+      const retirementValues: RetireFormValues = {
+        retiredAmount: values.amount,
+        note: values.note || '',
+        country: values.country,
+        stateProvince: values.stateProvince,
+        postalCode: values.postalCode,
+      };
+      errors = validateCreditRetire(
+        availableTradableAmount,
+        retirementValues,
+        errors,
+      );
+
+      // combo validation: send + retire
+      if (
+        Number(values.amount) + Number(values.amount) >
+        availableTradableAmount
+      ) {
+        errors.amount = insufficientCredits;
+      }
+    }
+    return errors;
+  };
+
+  const submitHandler = async (values: CreditTakeFormValues): Promise<void> => {
+    const msgTake: MsgTakeValues = {
+      owner: accountAddress,
+      basketDenom: basket.basketDenom,
+      amount: (values.amount * Math.pow(10, basket.exponent)).toString(),
+      retireOnTake: !!values.retireOnTake,
+      retirementLocation: values.retirementLocation,
+      retirementNote: values?.note,
+    };
+
+    onSubmit(msgTake);
+  };
+
+  const AutoSetRetirementLocation = (): JSX.Element => {
+    const {
+      values: { country, stateProvince, postalCode },
+      setFieldValue,
+    } = useFormikContext<CreditTakeFormValues>();
+
+    useEffect(() => {
+      const setRetirementLocation = async (): Promise<void> => {
+        const isoString = await getISOString(
+          country,
+          stateProvince,
+          postalCode,
+        );
+        setFieldValue('retirementLocation', isoString);
+      };
+
+      if (stateProvince || country || postalCode) {
+        setRetirementLocation();
+      }
+    }, [country, stateProvince, postalCode, setFieldValue]);
+
+    return <></>;
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validate={validateHandler}
+      onSubmit={submitHandler}
+    >
+      {({ values, submitForm, isSubmitting, isValid, submitCount, status }) => (
+        <Form>
+          <>
+            <AmountField
+              name="amount"
+              label="Amount"
+              availableAmount={availableTradableAmount}
+              batchDenom={basket.basketDenom}
+            />
+            <Field
+              component={CheckboxLabel}
+              type="checkbox"
+              name="retireOnTake"
+              className={styles.checkboxLabel}
+              label={
+                <Description className={styles.checkboxDescription}>
+                  Retire credits upon transfer
+                </Description>
+              }
+            />
```

Comment:
if the basket `disable_auto_retire` field is false then retirement is mandatory on take so in this case we should just check the retire checkbox by default and users should not be able to change that.

See:
- docs: https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgCreate 
- the **important note** in https://github.com/regen-network/regen-registry/issues/816
- design: https://www.figma.com/file/x5vjWsddiUBzP2N13AFOPw/Toucan-like-Flow-(Baskets)?node-id=628%3A15466

Reply:
Thanks! Done.